### PR TITLE
Support computing RNN-T loss with torchaudio

### PR DESCRIPTION
**Pull request by csukuangfj**
The motivation of this PR is to make it easier to compare the results using various implementations of RNN-T loss, e.g., torchaudio, k2 pruned RNN-T, and optimized_transducer.

### CODE REVIEW

**Comment by csukuangfj** [comment]
The training command is
```bash
export CUDA_VISIBLE_DEVICES="0,1,2,3,4,5,6,7"

./transducer_stateless2/train.py \
  --world-size 8 \
  --num-epochs 60 \
  --start-epoch 0 \
  --exp-dir transducer_stateless2/exp \
  --full-libri 1 \
  --max-duration 300 \
  --lr-factor 5
```
and the tensorboard log can be found at https://tensorboard.dev/experiment/1GOnjZqMTYCjMGRCKLJimA/

**Comment by csukuangfj** [comment]
Here are the results for this PR

|                                     | test-clean | test-other | comment                                                                        |
|-------------------------------------|------------|------------|--------------------------------------------------------------------------------|
| greedy search (max sym per frame 1) | 2.65       | 6.30       | --epoch 59 --avg 10  --max-duration 600                                        |
| greedy search (max sym per frame 2) | 2.62       | 6.23       | --epoch 59 --avg 10  --max-duration 100                                        |
| greedy search (max sym per frame 3) | 2.62       | 6.23       | --epoch 59 --avg 10  --max-duration 100                                        |
| modified beam search                | 2.63       | 6.15       | --epoch 59 --avg 10  --max-duration 100 --decoding-method modified_beam_search |
| beam search                         | 2.59       | 6.15       | --epoch 59 --avg 10  --max-duration 100 --decoding-method beam_search          |

Note: This PR uses torchaudio's RNN-T loss, which implements the standard RNN-T. It uses neither modified transducer nor pruned RNN-T. As a result, you can see that there is a degradation in WER when we limit the max symbol per frame to 1.

You can find the training logs, pre-trained model, decoding logs, and decoding results by visiting
https://huggingface.co/csukuangfj/icefall-asr-librispeech-transducer-stateless2-torchaudio-2022-04-19

---

Will also try it with the reworked model in a separate PR.
